cw-orch/src/daemon/core.rs

```diff
@@ -139,98 +126,78 @@ impl TxHandler for Daemon {
             funds: parse_cw_coins(coins)?,
         };
 
-        let result = self
-            .rt_handle
-            .block_on(sender.commit_tx(vec![init_msg], None))?;
-        // let address = &result.get_attribute_from_logs("instantiate", "_contract_address")[0].1;
+        let result = sender.commit_tx(vec![init_msg], None).await?;
 
         Ok(result)
     }
 
-    fn query<Q: Serialize + Debug, T: Serialize + DeserializeOwned>(
+    pub async fn query<Q: Serialize + Debug, T: Serialize + DeserializeOwned>(
         &self,
         query_msg: &Q,
         contract_address: &Addr,
     ) -> Result<T, DaemonError> {
         let sender = &self.sender;
         let mut client = cosmos_modules::cosmwasm::query_client::QueryClient::new(sender.channel());
-        let resp = self.rt_handle.block_on(client.smart_contract_state(
-            cosmos_modules::cosmwasm::QuerySmartContractStateRequest {
+        let resp = client
+            .smart_contract_state(cosmos_modules::cosmwasm::QuerySmartContractStateRequest {
                 address: contract_address.to_string(),
                 query_data: serde_json::to_vec(&query_msg)?,
-            },
-        ))?;
+            })
+            .await?;
 
         Ok(from_str(from_utf8(&resp.into_inner().data).unwrap())?)
     }
 
-    fn migrate<M: Serialize + Debug>(
+    pub async fn migrate<M: Serialize + Debug>(
         &self,
         migrate_msg: &M,
         new_code_id: u64,
         contract_address: &Addr,
-    ) -> Result<Self::Response, DaemonError> {
+    ) -> Result<CosmTxResponse, DaemonError> {
         let exec_msg: MsgMigrateContract = MsgMigrateContract {
             sender: self.sender.pub_addr()?,
             contract: AccountId::from_str(contract_address.as_str())?,
             msg: serde_json::to_vec(&migrate_msg)?,
             code_id: new_code_id,
         };
-        let result = self
-            .rt_handle
-            .block_on(self.sender.commit_tx(vec![exec_msg], None))?;
+        let result = self.sender.commit_tx(vec![exec_msg], None).await?;
         Ok(result)
     }
 
-    fn wait_blocks(&self, amount: u64) -> Result<(), DaemonError> {
-        let mut last_height = self
-            .rt_handle
-            .block_on(self.query::<Node>().block_height())?;
+    pub async fn wait_blocks(&self, amount: u64) -> Result<(), DaemonError> {
+        let mut last_height = self.query_client::<Node>().block_height().await?;
         let end_height = last_height + amount;
 
         while last_height < end_height {
-            // wait
-            self.rt_handle
-                .block_on(tokio::time::sleep(Duration::from_secs(4)));
+            tokio::time::sleep(Duration::from_secs(4)).await;
```

Comment:
There should be a way for us to get the average block time. Maybe wait for 2 blocks, get the average time and bump it by 20%? 

Because block times can vary a lot.

Reply:
see #125